bazel/generate_objc.bzl

```diff
@@ -0,0 +1,225 @@
+load(
+    "//bazel:protobuf.bzl",
+    "get_include_protoc_args",
+    "get_plugin_args",
+    "proto_path_to_generated_filename",
+)
+load(
+    ":grpc_util.bzl", 
+    "to_upper_camel_with_extension",
+    "label_to_file",
+)
+
+_GRPC_PROTO_HEADER_FMT = "{}.pbrpc.h"
+_GRPC_PROTO_SRC_FMT = "{}.pbrpc.m"
+_PROTO_HEADER_FMT = "{}.pbobjc.h"
+_PROTO_SRC_FMT = "{}.pbobjc.m"
+_GENERATED_PROTOS_DIR = "_generated_protos"
+
+_GENERATE_HDRS = 1
+_GENERATE_SRCS = 2
+_GENERATE_NON_ARC_SRCS = 3
+
+def _generate_objc_impl(ctx):
+    """Implementation of the generate_objc rule."""
+    protos = [
+        f
+        for src in ctx.attr.deps
+        for f in src[ProtoInfo].transitive_imports.to_list()
+    ]
+    outs = []
+
+    label_package = _join_directories([ctx.label.workspace_root, ctx.label.package])
```

Comment:
nit: This is not really label; this is the path to the package. Naming it `label_package` somehow makes the code hard to read, particularly in the `_strip_package_from_path` function.

Reply:
"label" here is this target's label (the one invoking this impl). Maybe rename it to `target_package`?